packages/react/src/components/TextArea/TextArea.tsx

```diff
@@ -226,19 +226,29 @@ const TextArea = frFn((props, forwardRef) => {
   }, [value, defaultValue, counterMode]);
 
   useIsomorphicEffect(() => {
-    const measuredWidth = wrapperRef.current?.scrollWidth;
     if (other.cols && textareaRef.current) {
       textareaRef.current.style.width = '';
       textareaRef.current.style.resize = 'none';
     } else if (textareaRef.current) {
       textareaRef.current.style.width = `100%`;
     }
-    [helperTextRef, errorTextRef, warnTextRef].forEach((r) => {
-      if (r.current) {
-        r.current.style.maxWidth = `${measuredWidth}px`;
-        r.current.style.overflowWrap = 'break-word';
-      }
+
+    if (!wrapperRef.current) return;
+    const applyWidth = (width: number) => {
+      [helperTextRef, errorTextRef, warnTextRef].forEach((r) => {
+        if (r.current) {
+          r.current.style.maxWidth = `${width}px`;
+          r.current.style.overflowWrap = 'break-word';
+        }
+      });
+    };
+
+    const resizeObserver = new ResizeObserver(([entry]) => {
+      applyWidth(entry.contentRect.width);
     });
+    resizeObserver.observe(wrapperRef.current);
+
+    return () => resizeObserver.disconnect();
```

Comment:
Does this need to be guarded in case it's not defined from the early return up on line 236?


```suggestion
    return () => resizeObserver && resizeObserver.disconnect();
```

Reply:
I don't think so because the observer is defined after the early return on line 236

Reply:
The cleanup function will always run on unmount

Reply:
Oh I see ok thanks, good catch!